course_catalog/models.py

```diff
@@ -55,6 +56,29 @@ def prefetch_list_items_for_user(self, user=None):
             Prefetch("list_items", queryset=UserListItem.objects.none())
         )
 
+    def prefetch_stafflist_items_for_user(self, user=None):
+        """Prefetch stafflist_items based on the current user"""
+        if (
+            user
+            and user.is_authenticated
+            and (
+                user.is_staff
+                or user.is_superuser
```

Comment:
**Question:** Do you think it's worth moving this computation to some util class? Similar to `is_public_list_editor(user)`. I think we'll need the same computation in the `infinite_example/views.py` file for the frontend SETTINGS object, too. *We should change the name of that file...*.

Also... https://github.com/mitodl/open-discussions/blob/8f92b2f9f9cf932c3df9a9494ce7064a8ad563c5/open_discussions/permissions.py#L28-L38 might be useful ?

Reply:
There are helper functions but like this one, they take `request` rather than `user` as input since they are always used in the context of a view (and there's no request here).  I think it would be useful to change the input from a HttpRequest to a User (pass in request.user instead of request) but that should probably be done in a separate PR since they are used pretty extensively in permissions.